pkg/registry/node/strategy.go

```diff
@@ -153,17 +153,30 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	}
 	host := hostIP.String()
 
-	if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
-		// Ignore requested scheme, use scheme provided by GetConnectionInfo
+	// We check if we want to get a default Kubelet's transport. It happens if either:
+	// - no port is specified in request (Kubelet's port is default),
+	// - we're using Port stored as a DaemonEndpoint and requested port is a Kubelet's port stored in the DaemonEndpoint,
+	// - there's no information in the API about DaemonEnpoint (legacy cluster) and requested port is equal to ports.KubeletPort (cluster-wide config)
+	defaultKubeletPort := node.Status.DaemonEndpoints.KubeletEndpoint.Port
+	if defaultKubeletPort == 0 {
+		defaultKubeletPort = ports.KubeletPort
+	}
+	if portReq == "" || strconv.Itoa(defaultKubeletPort) == portReq {
 		scheme, port, kubeletTransport, err := connection.GetConnectionInfo(host)
 		if err != nil {
 			return nil, nil, err
 		}
+		var portString string
+		if node.Status.DaemonEndpoints.KubeletEndpoint.Port != 0 {
```

Comment:
Ugh.. not sure. I can do it easily enough, but I'm not certain it'll be simpler to understand. Currently we have:
- set `defaultKubeletPort` to one stored in `KubeletEndpoint` or `ports.KubeletPort` if former is not set
- if `KubeletEndpoint` has a port set then use it, otherwise use the port returned by `GetConnectionInfo`

If I'm not mistaken it will always be `defaultKubeletPort`, but I need to make sure it is the case.